tests/interpreters/test_onnx_interpreter.py

```diff
@@ -0,0 +1,96 @@
+import pytest
+
+from xdsl.dialects import onnx
+from xdsl.dialects.builtin import ModuleOp, TensorType, f32
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.shaped_array import ShapedArray
+from xdsl.utils.test_value import TestSSAValue
+
+found_numpy = pytest.importorskip(
+    "numpy", reason="numpy is an optional dependency in xDSL"
+)
+import numpy as np  # noqa: I001, E402
+from xdsl.interpreters.onnx import OnnxFunctions  # noqa: I001, E402
+
+interpreter = Interpreter(ModuleOp([]))
+interpreter.register_implementations(OnnxFunctions())
+
+
+def test_onnx_add():
+    op = onnx.Add(
+        TestSSAValue(TensorType(f32, [2, 3])),
+        TestSSAValue(TensorType(f32, [2, 3])),
+        res_type=TensorType(f32, [2, 3]),
+    )
+
+    a = ShapedArray(np.array([1, 2, 3, 4, 5, 6]), [2, 3])
+    b = ShapedArray(np.array([1, 4, 2, 5, 3, 6]), [2, 3])
+    c = ShapedArray(np.array([0, 0, 0, 0, 0, 0]), [2, 3])
```

Comment:
nit: 
In my understanding, these don't need to be numpy arrays, changing them to Python lists would make this test a little more readable.
```suggestion
    a = ShapedArray([1, 2, 3, 4, 5, 6], [2, 3])
    b = ShapedArray([1, 4, 2, 5, 3, 6], [2, 3])
    c = ShapedArray([0, 0, 0, 0, 0, 0], [2, 3])
```